map/src/map/layers/CustomTileLayer.js

```diff
@@ -0,0 +1,283 @@
+                },
+            });
+        }
+        return res;
+    }
+
+    function getIconStyle(properties) {
+        let customIcon = new L.icon({
+            iconUrl: properties.iconUrl,
+            iconSize: [15, 15],
+            iconAnchor: [8.5, 8.5],
+        });
+        return {
+            icon: customIcon,
+            radius: 15,
+        };
+    }
+
+    function createIconLayerGroup(feature, latlng) {
+        const createMarker = (iconUrl, iconSize, offset = [0, 0]) => {
+            const icon = L.icon({
+                iconUrl,
+                iconSize,
+                iconAnchor: [iconSize[0] / 2, iconSize[1] / 2],
+                className: offset[0] !== 0 || offset[1] !== 0 ? styles.frontIcon : '',
+            });
+
+            const marker = L.marker(latlng, { icon });
+            marker.on('click', () => console.log('Clicked on', feature));
+            return marker;
+        };
+
+        const markers = [];
+
+        if (feature.properties.shield) {
+            const backIconUrl = `/map/images/${SHIELDS_FOLDER}/h_${feature.properties.shield}.svg`;
+            markers.push(createMarker(backIconUrl, [20, 20]));
+        }
+
+        const frontIconUrl = `/map/images/${MarkerOptions.POI_ICONS_FOLDER}/mx_${feature.properties.mainIcon}.svg`;
+        markers.push(createMarker(frontIconUrl, [12, 12], [0, 0]));
+
+        return L.layerGroup(markers);
+    }
+
+    function createTextLayerGroup(feature, latlng) {
+        const splitTextIntoLines = (text, maxLineLength = 15) => {
+            if (text.length > maxLineLength) {
+                const words = text.split(' ');
+                let lines = [''];
+                let currentLineIndex = 0;
+
+                words.forEach((word) => {
+                    if ((lines[currentLineIndex] + word + ' ').length <= maxLineLength) {
+                        lines[currentLineIndex] += `${word} `;
+                    } else {
+                        lines.push(`${word} `);
+                        currentLineIndex++;
+                    }
+                });
+
+                return lines;
+            } else {
+                return [text];
+            }
+        };
+
+        const createTextMarker = ({
+            text,
+            color = 'black',
+            fontSize = '12',
+            fontFamily = 'Arial',
+            textShadow = 0,
+            textShadowColor,
+            bold = false,
+            italic = false,
+        }) => {
+            const lines = splitTextIntoLines(text);
+            const shieldedLines = lines.map((line) => {
+                const style = `color: ${color}; text-shadow: ${textShadow}px ${textShadow}px 0px ${textShadowColor}; font-size: ${fontSize}px; font-family: ${fontFamily}; font-weight: ${bold ? 'bold' : 'normal'}; font-style: ${italic ? 'italic' : 'normal'}; white-space: nowrap;`;
+                return `<span class="shield" style="${style}">${line}</span>`;
```

Comment:
Is it ok that each `line` might contain space at the end (from splitTextIntoLines)? Use trim()?